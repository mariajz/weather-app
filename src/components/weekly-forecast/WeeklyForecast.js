import React from 'react';
import { ScrollView } from 'react-native';
import { StyledText } from '../../commons/styles';
import { CustomIcon } from '../../commons/visual-elements';
import DailySummary from './DailySummary';
import { CalenderWrapper, WeeklyForecastWrapper } from './WeeklyForecast.style';

const WeeklyForecast = ({ weeklyForecastProps }) => {
    const { forecastData } = weeklyForecastProps;

    return (
        <WeeklyForecastWrapper>
            <CalenderWrapper>
                <CustomIcon iconName="calendar" />
                <StyledText marginLeft={5} size={12}>
                    14-DAY FORECAST
                </StyledText>
            </CalenderWrapper>
            <ScrollView showsVerticalScrollIndicator={false}>
                {forecastData.map(item => (
                    <DailySummary dailyWeatherItem={item} key={item.day} />
                ))}
            </ScrollView>
        </WeeklyForecastWrapper>
    );
};

export default WeeklyForecast;
