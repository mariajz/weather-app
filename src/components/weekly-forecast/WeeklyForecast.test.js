import { render } from '@testing-library/react-native';
import React from 'react';
import WeeklyForecast from './WeeklyForecast';

const defaultProps = {
    weeklyForecastProps: {
        forecastData: [
            {
                chanceOfRain: 0,
                condition: 11,
                day: '2023-08-14',
                maxTemp: 30.8,
                minTemp: 21.1,
                willItRain: 0,
            },
            {
                chanceOfRain: 89,
                condition: 9,
                day: '2023-08-15',
                maxTemp: 28.9,
                minTemp: 20.5,
                willItRain: 1,
            },
        ],
    },
};
describe('WeeklyForecast', () => {
    it('should render WeeklyForecast', () => {
        const container = render(<WeeklyForecast {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
