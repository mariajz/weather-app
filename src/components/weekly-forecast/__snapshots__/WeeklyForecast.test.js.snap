// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyForecast should render WeeklyForecast 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "display": "flex",
      "flexDirection": "column",
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/icons/calendar.png",
        }
      }
      style={
        {
          "height": 25,
          "width": 25,
        }
      }
      testID="icon"
      tintColor="white"
    />
    <Text
      marginLeft={5}
      size={12}
      style={
        {
          "color": "white",
          "fontSize": 12,
          "marginLeft": 5,
        }
      }
    >
      14-DAY FORECAST
    </Text>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "borderColor": "grey",
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "marginTop": 5,
            "width": "100%",
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-between",
            "paddingBottom": 6,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 6,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginLeft": 0,
            }
          }
        >
          2023-08-14
        </Text>
        <View>
          <Image
            height={20}
            source={11}
            style={
              {
                "height": 20,
                "width": 20,
              }
            }
            testID="image"
            width={20}
          />
        </View>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginLeft": 0,
            }
          }
        >
          21.1
          째
          C
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginLeft": 0,
            }
          }
        >
          30.8
          째
          C
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "grey",
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "marginTop": 5,
            "width": "100%",
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-between",
            "paddingBottom": 6,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 6,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginLeft": 0,
            }
          }
        >
          2023-08-15
        </Text>
        <View>
          <Image
            height={20}
            source={9}
            style={
              {
                "height": 20,
                "width": 20,
              }
            }
            testID="image"
            width={20}
          />
          <Text
            style={
              {
                "color": "blue",
                "fontSize": 10,
                "marginLeft": 0,
              }
            }
          >
            89
            %
          </Text>
        </View>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginLeft": 0,
            }
          }
        >
          20.5
          째
          C
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontSize": 12,
              "marginLeft": 0,
            }
          }
        >
          28.9
          째
          C
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
